torch/distributed/tensor/_utils.py

```diff
@@ -271,6 +271,70 @@ def compute_global_tensor_info(
     return tensor_shape, tensor_stride
 
 
+def compute_local_tensor_info(
+    global_tensor: torch.Tensor,
+    mesh: DeviceMesh,
+    placements: Sequence[Placement],
+) -> tuple[list[int], list[int]]:
+    """
+    Compute the local size and stride of a DTensor from the given global tensor info.
+
+    For example, if we have a global tensor with size (4, 8, 4) and stride (32, 1, 8).
+    If the DTensor placements are [Shard(2)] and world_size is 2;
+    then the local size is (4, 8, 2) and stride is (16, 1, 8).
```

Comment:
It seems worth pointing out this only works for even sharding. Because its companion compute_global_tensor_info obviously only works for even sharding (since you only have the local info), but this function COULD in principle work for uneven.